### PR TITLE
Ticket-547 updated sub-user service and controller to use user entity

### PR DESCRIPTION
**Pull request by ckirby19**
### Issue link / number:

#547 

### What changes did you make?

Replace use of req["user"] with req["userEntity"] and addressed other issues where needed

### Why did you make the changes?

As requested in the ticket as part of a larger refactor

### Did you run tests?

Yes, unit and e2e tests

### CODE REVIEW

File: src/subscription-user/subscription-user.controller.ts [hunk 1]
```diff
@@ -44,8 +45,8 @@ export class SubscriptionUserController {
     @Body() updateSubscriptionsDto: UpdateSubscriptionUserDto,
   ): Promise<ISubscriptionUser | undefined> {
     return this.subscriptionUserService.cancelWhatsappSubscription(
-      req['user'].user.id,
-      req['user'].user.email,
+      req['userEntity'].user.id,
+      req['userEntity'].user.email,
```
Comment (annarhughes) [inline src/subscription-user/subscription-user.controller.ts:49]:
```suggestion
      req['userEntity'].id,
      req['userEntity'].email,
```

the new `userEntity` object doesn't have a child `user` object

**Review by annarhughes — CHANGES_REQUESTED** [review]
Thanks for addressing this issue @ckirby19 🙏 One type issue to fix and it's ready to be merged!

File: src/subscription-user/subscription-user.controller.ts (hunk 1 above)
Comment (ckirby19) [reply src/subscription-user/subscription-user.controller.ts]: Ahh yes apologies, I have changed that now

**Review by eleanorreem — APPROVED** [review]

**Comment by eleanorreem** [comment]
Thanks @ckirby19 looks great 😄

**Pull request merged**